xskillscore/core/deterministic.py

```diff
@@ -95,9 +118,11 @@ def rmse(a, b, dim):
     xarray.apply_ufunc
 
     """
+    dim, axis = _preprocess(dim)
+
     return xr.apply_ufunc(_rmse, a, b,
-                          input_core_dims=[[dim], [dim]],
-                          kwargs={'axis': -1},
+                          input_core_dims=[dim, dim],
```

Comment:
Is this suppose to be 'new_dim'?
I'm curious why you use `dim` and `axis` as input into `rmse` but `new_dim` and `-1` as input to `pearson_r`.

Reply:
I think the main difference between pearsonr and rmse/mse/mae is that rmse/mse/mae simply can pass in the multiple axes you want to perform the operation
`res = np.sqrt(((a - b) ** 2).mean(axis=axis))`

pearsonr does it implicitly summing only axis=0 so I stack the other dimensions into one

Reply:
In essence, what this multi-dim thing is doing is:

```
a = xr.DataArray([[0, 1, 2, 3], [4, 5, 6, 7]])
b = xr.DataArray([[4, 5, 6, 8], [7, 8, 9, 10]])

from scipy.stats import pearsonr
pearsonr(a.values.ravel(), b.values.ravel())
(0.9619647660338173, 0.00013366763860081336)

from xskillscore import pearson_r
xskillscore.pearson_r(a, b, ['dim_0', 'dim_1'])
<xarray.DataArray ()>
array(0.961965)
```